kuksa-persistence-provider/Readme.md

```diff
@@ -0,0 +1,100 @@
+
+## Configuration: config.json
+
+Main configuration is in config.json, and example may look like this
+
+```json
+{
+    "restore-only": {
+        "values": [
+            "Vehicle.VehicleIdentification.VIN",
+            "Vehicle.VehicleIdentification.VehicleInteriorColor"
+        ],
+        "actuators": [
+            "Vehicle.Cabin.Infotainment.HMI.TemperatureUnit"
+        ]
+    },
+
+    "restore-and-watch": {
+        "values": [
+            "Vehicle.Cabin.Infotainment.HMI.TemperatureUnit",
+            "Vehicle.Cabin.HVAC.Station.Row4.Passenger.FanSpeed"
+        ],
+        "actuators": [
+            "Vehicle.Cabin.Infotainment.Media.Volume"
+        ]
+    },
+    "state-storage": {
+        "type": "file",
+        "path": "statestore.json"
+    }
+}
+```
+
+### section restore-only
+
+These elements will be restored from the state store upon startup, but their values will not be watched and updated for changes. You can define whether the current values (values) will be restored or whether a target value is set (actuators).
+
+### section restore-and-watch
+
+These elements will be restored from the state store upon startup. It is the intention to also monitor their state and update it in the state store. You can define whether the current values (values) will be restored and watched or whether a target value is set (actuators) and watched. As restore-and-watch includes restore, there is no need to add paths in restore-and-watch to restore-only as well.
+
+## state-storage
+
+Configures the state sotrage used to retrieve values. Currently supported: file
+
+## File storage: statestore.json
+
+This is a valid state store for the file storage.
+*Note: ALl VALUES NEED TO BE STORED AS STRING*.
+As the statestore does not make a difference between current and target (actuation) values it is currently not possible to watch or restore both for a single VSS path.
+
+```json
+
+{
+    "Vehicle.VehicleIdentification.VIN": {
+        "value": "DEADBEEF"
+    },
+    "Vehicle.VehicleIdentification.VehicleInteriorColor": {
+        "value": "Black"
+    }
+}
+```
+
+## Features
+
+* djson: Use a dependable json library.
+  Check out the repo with the depenable json library at the same level as the parent folder or project folder of this repo.
+  Enable with ```cargo build --features json_djson --no-default-features```
+
+
+## Build
+
+```bash
+cargo build --features json_djson --no-default-features
+```
+
+## Test
+
+Start kuksa databroker with:
+
+```bash
+docker run -it --rm --net=host ghcr.io/eclipse-kuksa/kuksa-databroker:latest --port 55556
+```
+
+Start kuksa sdk with:
+
+```bash
+docker run -it --rm --net=host ghcr.io/eclipse-kuksa/kuksa-python-sdk/kuksa-client:latest grpc://127.0.0.1:55556
+# Check Vales in cmd
+getValue Vehicle.Cabin.HVAC.Station.Row4.Passenger.FanSpeed
+```
```

Comment:
```suggestion

1) Check for empty data point after startup
   
    * In Terminal A): Start kuksa databroker with:

        ```bash
        docker run -it --rm --net=host ghcr.io/eclipse-kuksa/kuksa-databroker:latest --port 55556
        ```

    * In Terminal B) Start kuksa sdk with:

        ```bash
        docker run -it --rm --net=host ghcr.io/eclipse-kuksa/kuksa-python-sdk/kuksa-client:latest grpc://127.0.0.1:55556
        # Check Vales in cmd
        getValue Vehicle.Cabin.HVAC.Station.Row4.Passenger.FanSpeed
        ```

    * In Terminal C) Start kuksa persistency provider with:

        ```bash
        cargo run
        ```

2) Set data point which is stored by persistency provider
    * In Terminal B) :

        ```bash
        # Set Vales in cmd
        setValue Vehicle.Cabin.HVAC.Station.Row4.Passenger.FanSpeed 40
        ```
3) Restart and check if datapoint with old value is restored by persistency provider
```